x/tokenregistry/keeper/keeper.go

```diff
@@ -47,55 +47,106 @@ func (k keeper) CheckEntryPermissions(entry *types.RegistryEntry, requiredPermis
 	return true
 }
 
-func (k keeper) GetEntry(wl types.Registry, denom string) (*types.RegistryEntry, error) {
-	for i := range wl.Entries {
-		e := wl.Entries[i]
-		if e != nil && e.Denom == denom {
-			return wl.Entries[i], nil
-		}
-	}
-	return nil, errors.Wrap(errors.ErrKeyNotFound, "registry entry not found")
+func (k keeper) getDenomPrefix(ctx sdk.Context, denom string) []byte {
+	return append(types.TokenDenomPrefix, []byte(denom)...)
 }
 
+// SetToken add a new denom
 func (k keeper) SetToken(ctx sdk.Context, entry *types.RegistryEntry) {
-	wl := k.GetRegistry(ctx)
-	for i := range wl.Entries {
-		if wl.Entries[i] != nil && types.StringCompare(wl.Entries[i].Denom, entry.Denom) {
-			wl.Entries[i] = entry
-			k.SetRegistry(ctx, wl)
-			return
-		}
-	}
-	wl.Entries = append(wl.Entries, entry)
-	k.SetRegistry(ctx, wl)
+	// get a copy to avoid modify input
+	tmpCopy := *entry
+	tmpCopy.Sanitize()
```

Comment:
We added this behavior to remove leading and trailing '/' in an ibc denom. @Jedi2002 @canercandan is that ok?